numpy/lib/tests/test_ufunclike.py

```diff
@@ -55,6 +55,10 @@ def __array_wrap__(self, obj, context=None):
                 obj.metadata = self.metadata
                 return obj
 
+            def __array_finalize__(self, obj):
+                self.metadata = getattr(obj, 'metadata', None)
```

Comment:
Can you describe in the commit message why these changes were needed?

Reply:
Done. This was indeed a non-trivial one - I had to look again to remind me why it was needed.  I wondered about the use of stating it in the commit message until I recalled that with `git blame` one can find why this was added.